spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/batch/JdbcSearchableJobExecutionDao.java

```diff
@@ -150,6 +178,52 @@ protected long getNextKey() {
 
 	}
 
+	@Override
+	public List<JobExecution> findJobExecutions(JobInstance job) {
+		Assert.notNull(job, "Job cannot be null.");
+		Assert.notNull(job.getId(), "Job Id cannot be null.");
+
+		final String sqlQuery = batchVersion.equals(BatchVersion.BATCH_4) ? FIND_JOB_EXECUTIONS_4 : FIND_JOB_EXECUTIONS_5;

```

Comment:
Please don't use `final` everywhere. The consensus we reached was to use it for immutable instance vars.